### PR TITLE
housekeeping: Migrate AccountMenu tests to utilize a page object

### PR DESCRIPTION
**Pull request by yhabib**
# Motivation

We want to be consistent in how we test components using page objects.

# Changes

- Migrate `AccountMenu` tests to only use PageObjects

# Tests

- They work as before

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary

### CODE REVIEW

File: frontend/src/tests/page-objects/AccountMenu.page-object.ts [hunk 1]
```diff
@@ -11,6 +11,11 @@ export class AccountMenuPo extends BasePageObject {
     return new AccountMenuPo(element.byTestId(AccountMenuPo.TID));
   }
 
+  isOpen(): Promise<boolean> {
+    const popover = this.root.querySelector('[role="menu"]');
```
Comment (yhabib) [inline frontend/src/tests/page-objects/AccountMenu.page-object.ts:15]: I could have created a PO for this component, but it felt like too much for just one assertion.

**Review by mstrasinskis — APPROVED** [review]
Thank you!

File: frontend/src/tests/page-objects/AccountMenu.page-object.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/tests/page-objects/AccountMenu.page-object.ts:15]: Fine by me.

File: frontend/src/tests/lib/components/header/AccountMenu.spec.ts [hunk 2]
```diff
@@ -48,33 +43,31 @@ describe("AccountMenu", () => {
     });
 
     it("should be open", async () => {
-      const renderResult = render(AccountMenu);
-
-      await show(renderResult);
+      // const renderResult = render(AccountMenu);
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/header/AccountMenu.spec.ts:46]: Should this be here?

File: frontend/src/tests/page-objects/AccountMenu.page-object.ts [hunk 3]
```diff
@@ -26,6 +31,34 @@ export class AccountMenuPo extends BasePageObject {
...
 
+  getSignInButtonPo(): LinkPo {
+    return LinkPo.under({
+      element: this.root,
+      testId: "toolbar-login",
+    });
+  }
+
+  getSignOutButtonPo(): LinkPo {
+    return LinkPo.under({
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/AccountMenu.page-object.ts:42]: The Logout button is an actual button:
https://github.com/dfinity/nns-dapp/blob/361324e872a1a39465def1e30eda74bb4c825355/frontend/src/lib/components/header/Logout.svelte#L15

So using `LinkPo` works by pure coincidence. Let's use `ButtonPo` here to be consistent with the actual DOM.

File: frontend/src/tests/page-objects/AccountMenu.page-object.ts [hunk 4]
```diff
@@ -26,6 +31,34 @@ export class AccountMenuPo extends BasePageObject {
     });
   }
 
+  getSignInButtonPo(): LinkPo {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/AccountMenu.page-object.ts:34]: The Login button is an actual button:
https://github.com/dfinity/nns-dapp/blob/361324e872a1a39465def1e30eda74bb4c825355/frontend/src/lib/components/header/LoginIconOnly.svelte#L8
So using `LinkPo` works by pure coincidence. Let's use `ButtonPo` here to be consistent with the actual DOM.

File: frontend/src/tests/page-objects/AccountMenu.page-object.ts [hunk 5]
```diff
@@ -26,6 +31,34 @@ export class AccountMenuPo extends BasePageObject {
...
+  }
+
+  getSignOutButtonPo(): LinkPo {
+    return LinkPo.under({
+      element: this.root,
+      testId: "logout",
+    });
+  }
+
+  getSettingsButtonPo(): LinkPo {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/AccountMenu.page-object.ts:48]: We have a page object called `ButtonPo` so with this method name it's a bit confusing that it doesn't return ` ButtonPo`.
How about `getSettingsButtonLinkPo` or `getSettingsLinkPo`?

Same below.

Comment (yhabib) [reply frontend/src/tests/page-objects/AccountMenu.page-object.ts:48]: I've changed to `getSettingsButtonPo` as it is a Button. I'll check in another one if we can refactor the component into a Link and then rename these ones.

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/page-objects/AccountMenu.page-object.ts (hunk 1 above)
Comment (yhabib) [reply frontend/src/tests/page-objects/AccountMenu.page-object.ts:16]: indeed. much cleaner 🙌

**Pull request merged**